parser/inline.go

```diff
@@ -1192,9 +1192,6 @@ func helperEmphasis(p *Parser, data []byte, c byte) (int, ast.Node) {
 
 	for i < len(data) {
 		length := helperFindEmphChar(data[i:], c)
-		if length == 0 {
```

Comment:
Building on last commit, in the triple `c` case, we'd still have short-circuited here - but we actually want to claim this `c` as our own.  So we do the rest of the validity checks first, and then do this check at the end to exit the loop